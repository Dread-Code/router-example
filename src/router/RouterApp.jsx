import React from "react";
import { AboutScreen } from "../components/AboutScreen/AboutScreen";
import { HomeScreen } from "../components/HomeScreen/HomeScreen";
import { LoginScreen } from "../components/LoginScreen/LoginScreen";
import { Switch, Route } from "react-router-dom";

export const RouterApp = () => {
  return (
    <div>
      {/* A <Switch> looks through its children <Route>s and
          renders the first one that matches the current URL. */}
      <Switch>
        <Route path="/about" component={AboutScreen} />
        <Route path="/login" component={LoginScreen} />
        <Route path="/" component={HomeScreen} />
      </Switch>
    </div>
  );
};
