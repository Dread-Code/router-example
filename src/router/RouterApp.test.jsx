import React from "react";
import { mount } from "enzyme";
import { BrowserRouter as Router } from "react-router-dom";
import { UserContext } from "../context/UserContext";
import { RouterApp } from "./RouterApp";

describe("RouterApp", () => {
  const setUser = jest.fn();
  const user = {
    id: 123,
    name: "LUcas",
    email: "LUCAS@GMAIL.COM",
  };
  const wrapper = mount(
    <UserContext.Provider value={{ user, setUser }}>
      <Router>
        <RouterApp />
      </Router>
    </UserContext.Provider>
  );
  test("should match snapshoot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
