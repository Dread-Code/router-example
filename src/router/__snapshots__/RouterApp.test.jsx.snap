// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouterApp should match snapshoot 1`] = `
<div
  className="container"
>
  <div>
    <h1>
      HomeScreen
    </h1>
    <pre>
      {"id":123,"name":"LUcas","email":"LUCAS@GMAIL.COM"}
    </pre>
  </div>
</div>
`;
